### ISSUE TITLE
using follow twice on same JSON resource fails

### ISSUE THREAD

**Issue by x1B** [alex3683/hal-http-client#15]
Using the *following* 😉 code 

```js
hal.follow( entry, 'scenarios' )
   .on( { '200': hal.thenFollowAll( 'item' ) } );
hal.follow( entry, 'scenarios' )
   .on( { '200': hal.thenFollowAll( 'item' ) } );
```

fails for me (it works when performing only a single call).

I get the *following* error:

```
hal-http-client.js:748 Uncaught (in promise) TypeError: Already read
    at hal-http-client.js:748
    at <anonymous>
  | (anonymous) | @ | hal-http-client.js:748
  | Promise rejected (async) |   |  
  | (anonymous) | @ | hal-http-client.js:748
  | Promise (async) |   |  
  | doFetch | @ | hal-http-client.js:899
  | get | @ | hal-http-client.js:224
  | (anonymous) | @ | hal-http-client.js:467
  | follow | @ | hal-http-client.js:431
  | ....
```

**Comment by alex3683** [comment]
Most probably this is caused by caching the fetch promise on GET (https://github.com/aixigo/hal-http-client/blob/master/hal-http-client.js#L173-L176). When using a cached promise we need to make sure that `json()` or `text()` are not called more than once on the fetch response object.

**Comment by alex3683** [comment]
Fixed on `master` (`0.8.0`).
